public/js/graph/react_node.js.jsx

```diff
@@ -0,0 +1,113 @@
+function renderReactGraph() {
+    React.render(
+        <ReactSVG width={1195} height={650}/>,
+        document.getElementById('react-graph')
+    );
+}
+
+function getAttributes(node_named_map) {
+    var attrs = [];
+    //Traversing a NodeNamedMap type
+    Array.prototype.slice.call(node_named_map).forEach(function(item) {
+        attrs[item.name] = item.value;
+    });
+    return attrs;
+}
+
+function getStyles(styles_strings) {
+    if (!styles_strings){
+        return {};
+    }
+    var styles = {};
+        styles_strings.split(";").map(function(key, value){
+        if (key){
+            styles[key.substring(0, key.indexOf(':'))] = key.substring(key.indexOf(':')+1);
+        }
+    });
+    return styles;
+}
+
+var ReactSVG = React.createClass({
+    render: function() {
+        return (
+            <svg width={this.props.width} height={this.props.height}>
+                {/*<ReactRegions/>*/}
+                <ReactNodes/>
+                {/*<ReactRegions/>*/}
+                {/*<ReactBools/>*/}
+                {/*<ReactEdges/>*/}
+                {/*<ReactRegionLabels/>*/}
+            </svg>
+        );
+    }
+});
+
+var ReactNodes = React.createClass({
+    getInitialState: function(){
+        return {
+            nodes_list: []
+        };
+    },
+    
+    componentDidMount: function(){
+        var dict_list = [];
+        
+        $('.node').map(function(key, value) {
+            var entry = {};
+            entry["attributes"] = getAttributes(value.attributes);
+            //assume no styles here
+            entry["style"] = getStyles(entry["attributes"]["style"]);
+            entry["children"] = [];
+            //value.children is an HTML collection, converting to array here
+            var children_arr = [].slice.call(value.children);
```

Comment:
I learned to do this from here:
http://stackoverflow.com/questions/222841/most-efficient-way-to-convert-an-htmlcollection-to-an-array
Doing `Array(value.children)` would give me an `[HTMLCollection[2]]`, but I wanted [rect, text] instead.